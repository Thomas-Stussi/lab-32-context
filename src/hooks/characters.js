import { useEffect, useState } from 'react';
import { getCharacters } from '../services/heyArnoldApi';

export const useCharacters = () => {
  const [characters, setCharacters] = useState([]);

  useEffect(() => {
    getCharacters()
      .then(fetchedCharacters => setCharacters(fetchedCharacters));
  }, []);

  return characters;
};
