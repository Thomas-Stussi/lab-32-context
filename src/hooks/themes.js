import { useContext } from 'react';
import { ThemeContext } from '../context/ThemeContext';
import styles from '../components/theme-layer/Theme.css';

export const useThemePicker = () => {
  const { theme, toggle } = useContext(ThemeContext);
  const className = styles[theme];

  return { theme, toggle, className };
};
