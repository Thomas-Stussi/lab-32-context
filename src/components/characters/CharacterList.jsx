/* eslint-disable max-len */
import React from 'react';
import { useCharacters } from '../../hooks/characters';
import { useThemePicker } from '../../hooks/themes';
import CharacterItem from './CharacterItem';
import styles from './CharacterList.css';
import themes from '../theme-layer/Theme.css';

const CharacterList = () => {
  const { theme } = useThemePicker();
  const characters = useCharacters();

  const characterElements = characters.map(character => (
    <li key={character.id}>
      <CharacterItem {...character} />
    </li>
  ));

  return (
    <ul data-testid="characters" className={`${styles.CharacterList} ${theme === 'dark' ? themes.dark : themes.light}`}>
      {characterElements}
    </ul>
  );
};

export default CharacterList;
