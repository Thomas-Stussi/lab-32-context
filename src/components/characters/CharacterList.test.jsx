/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';
import { getCharacters } from '../../services/heyArnoldApi';
import { useThemePicker } from '../../hooks/themes';

jest.mock('../../services/heyArnoldApi.js');
jest.mock('../../hooks/themes');

describe('CharacterList component', () => {
  afterEach(() => cleanup());
  it('renders CharacterList', () => {
    useThemePicker.mockResolvedValue('dark');
    getCharacters.mockResolvedValue([
      {
        '_id': '5da237699734fdcb7bef8f51',
        'name': 'Arnold Shortman',
        'image': 'https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333'
      }
    ]);
    const { asFragment } = render(<CharacterList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
