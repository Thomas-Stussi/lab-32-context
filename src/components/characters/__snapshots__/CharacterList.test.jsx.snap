// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<DocumentFragment>
  <ul
    class="CharacterList undefined"
    data-testid="characters"
  />
</DocumentFragment>
`;
